packages/base/card-api.gts

```diff
@@ -2369,9 +2369,13 @@ export function getFields(
           maybeFieldName
         );
         if (
-          opts?.usedFieldsOnly &&
+          opts?.usedLinkedFieldsOnly &&
           !usedFields.includes(maybeFieldName) &&
-          !maybeField?.isUsed
+          !maybeField?.isUsed &&
+          // Do not skip computed `contains` and `containsMany` fields.
+          !((maybeField?.fieldType === 'contains' ||
+            maybeField?.fieldType === 'containsMany') &&
+            maybeField?.computeVia)
```

Comment:
why the `computeVia` check? whether or not the field is a computed should not have any bearing on whether to consider it or not

Reply:
in fact we probably even want a test to prove that (a used computed linksTo and a used computed linksToMany field would be in teh search doc)

Reply:
If we also return unused non-computed `contains` or `containsMany` fields, it might lead to an infinite loop in `recompute.__loadModel` function

Reply:
so we should not have a card that has a contains or contains many that includes itself. We don't have any static way to discover that and alert the user (but we should). barring that situation (which should hopefully not exist in our tests), can you describe how we get into the infinite loop scenario?